### PR TITLE
Qute - remove standalone lines by default

### LINKED ISSUE #8753: Qute: remove whitespace around type declarations
If I declare parameter types in my template:

```txt
{@java.lang.String name}
{@java.lang.String foo}

Here is my email
```

Then the output is `\n\n\nHere is my email` because whitespace isn't stripped between the type declarations and the first content. Note that this is mostly an issue for email and plaintext, but still.

Now, perhaps it's important to allow whitespace before the first content, but even in this case, we should strip whitespace before/between type declarations and the last newline after it?

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #8753

### CODE REVIEW

**Comment by mkouba** [comment]
CC @adrianfiedler @FroMage @ia3andy

**Comment by ia3andy** [comment]
2 fast 2 furious @mkouba 😂 🍻 that's awesome... I'll have a look on monday if you don't mind?

**Comment by mkouba** [comment]
Ah, I forgot that `String.isBlank()` is `@since 11`...

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -124,6 +124,55 @@ A closing curly bracket (`}`) is ignored if not inside an expression/tag.
 
 TIP: It is also possible to use escape sequences `\{` and `\}` to insert delimiters in the text. In fact, an escape sequence is usually only needed for the start delimiter, ie. `\\{foo}` will be rendered as `{foo}` (no evaluation will happen).
 
+=== Removing Standalone Lines From the Template
+
+By default, the parser removes standalone lines from the template output.
+A *standalone line* is a line that contains only section tags (e.g. `{#each}` and `{/each}`), parameter declarations (e.g. `{@org.acme.Foo foo}`) and whitespace characters.
+The default behavior can be disabled by setting the property `quarkus.qute.remove-standalone-lines` to `false`.
```
Comment (jaikiran) [inline docs/src/main/asciidoc/qute-reference.adoc:131]: I think adding a sentence mentioning, what disabling such a behaviour can result in, would be a good thing. I see that there's an example output a few lines down, which shows what happens, but explicitly stating something like  "If this property is disabled, then each standalone line will result in a new line character, in the generated output" might be helpful

Comment (mkouba) [reply docs/src/main/asciidoc/qute-reference.adoc:131]: Yep, that makes sense. Except that it's not "a new line character" but "a bunch of whitespace chars resulting in a new line" ;-).

**Comment by ia3andy** [comment]
I have an error, I am rebuilding everything to make sure it's not due to my env..

**Comment by mkouba** [comment]
@ia3andy no problem. Let me know if you hit a problem...

File: independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java [hunk 2]
```diff
@@ -180,6 +180,18 @@ public void testCdata() {
         assertEquals("Hello world <strong>", engine.parse("Hello {name} {[<strong>]}").data("name", "world").render());
     }
 
+    @Test
+    public void testRemoveStandaloneLines() {
+        Engine engine = Engine.builder().addDefaults().removeStandaloneLines(true).build();
+        String content = "{@java.lang.String foo}\n" // -> standalone
+                + "\n"
```
Comment (FroMage) [inline independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java:187]: So, this appears to validate that the description:

> A *standalone line* is a line that contains only section tags (e.g. `{#each}` and `{/each}`), parameter declarations (e.g. `{@org.acme.Foo foo}`) and whitespace characters.

has a confusing `and`. I initially thought this applied to empty lines as well, but this test says it does not.

Perhaps the following is less confusing for people like me?

> A *standalone line* is a line that contains at least one of: section tags (e.g. `{#each}` and `{/each}`) or parameter declarations (e.g. `{@org.acme.Foo foo}`), ignoring any whitespace characters on the same line.

It's not perfect yet, though so if you find better, I'll take it :)

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 3]
```diff
@@ -124,6 +124,56 @@ A closing curly bracket (`}`) is ignored if not inside an expression/tag.
...
+     {#for item in items} <1>
+       <li>{item.name}</li> <2>
+     {/for} <3>
+     </ul>
+   <body>
+</html>
+----
+<1> This is a standalone line and will be removed.
+<2> This is not a standalone line as it contains non-whitespace characters and an expression.
+<3> This is a standalone line and will be removed.
```
Comment (FroMage) [inline docs/src/main/asciidoc/qute-reference.adoc:149]: Here I'm missing an example of a line with whitespace but no section/parameter, to show that they get kept.

**Review by FroMage — CHANGES_REQUESTED** [review]
LGTM except docs which can be confusing to people like me.

**Review by ia3andy — APPROVED** [review]
Very cool !!

The only thing to fix would be with the include inner blocks (which still generate new lines) as we discussed on zulip

File: independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java (hunk 2 above)
Comment (mkouba) [reply independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java:187]: Yeah, as I read it it's confusing... let's fix it!

Comment (mkouba) [reply independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java:187]: Is it better now?

**Review by FroMage — APPROVED** [review]
Yup, clearer, thanks!

**Comment by ia3andy** [comment]
@mkouba all good no more line  with the include inner block 🍻

**Comment by ia3andy** [comment]
@mkouba ready to merge :)

**Pull request merged**